### ISSUE TITLE
Test System.Net.Http.Functional.Tests.DiagnosticsTest.SendAsync_HttpTracingEnabled_Succeeds failing

### ISSUE THREAD

**Issue by steveharter** [dotnet/corefx#20199]
The outerrun test System.Net.Http.Functional.Tests.DiagnosticsTest.SendAsync_HttpTracingEnabled_Succeeds is failing with
```
MESSAGE:
Assert.Equal() Failure\nExpected: 42\nActual: 134
+++++++++++++++++++
STACK TRACE:
at System.Diagnostics.RemoteExecutorTestBase.RemoteInvokeHandle.Dispose() in /mnt/resource/j/workspace/dotnet_corefx/master/outerloop_netcoreapp_debian8.4_release/src/Common/tests/System/Diagnostics/RemoteExecutorTestBase.cs:line 208
```

It looks like a recent regression. It is likely failing on all Linux distros but currently just these are failing (probably because they ran recently):
- Debian 8 Release
- openSUSE 42.1 Release
- Red Hat 7.2 Debug
- Ubuntu 16.10 Debug
- PortableLinux Debug

**Comment by Jiayili1** [comment]
This issue is repro on Portable Core Tests, detail: https://mc.dot.net/#/product/netcore/master/source/official~2Fcorefx~2Fmaster~2F/type/test~2Ffunctional~2Fportable~2Fcli~2F/build/20170524.01/workItem/System.Net.Http.Functional.Tests

**Comment by v-haren** [comment]
failed again here: https://ci.dot.net/job/dotnet_corefx/job/master/job/outerloop_netcoreapp_centos7.1_debug/55/testReport/System.Net.Http.Functional.Tests/DiagnosticsTest/SendAsync_HttpTracingEnabled_Succeeds/

**Comment by Jiayili1** [comment]
This issue is also failed on Portable Core Tests (2.0.0), detail: https://mc.dot.net/#/product/netcore/200/source/official~2Fcorefx~2Frelease~2F2.0.0~2F/type/test~2Ffunctional~2Fportable~2Fcli~2F/build/20170524.01/workItem/System.Net.Http.Functional.Tests/analysis/xunit/System.Net.Http.Functional.Tests.DiagnosticsTest~2FSendAsync_HttpTracingEnabled_Succeeds

**Comment by stephentoub** [comment]
I'll take a look..

**Comment by stephentoub** [comment]
From the stack in the log, it looks like it's hitting some kind of fail fast in the runtime:
```
  FAULT_STACK:
  libc.so.6!gsignal
  libc.so.6!abort
  libcoreclr.so!UNKNOWN
  libcoreclr.so!UNKNOWN
  libcoreclr.so!CallDescrWorkerInternal
  libcoreclr.so!DispatchCallDebuggerWrapper(CallDescrData*, ContextTransitionFrame*, int)
  libcoreclr.so!DispatchCallSimple(unsigned long*, unsigned int, unsigned long, unsigned int)
  libcoreclr.so!MethodTable::RunClassInitEx(Object**)
  libcoreclr.so!MethodTable::DoRunClassInitThrowing()
  libcoreclr.so!JIT_GetSharedNonGCStaticBase_Helper
  System.Net.Http.dll!System.Net.Http.CurlHandler..ctor()
  System.Net.Http.dll!System.Net.Http.HttpClientHandler..ctor()
  System.Net.Http.dll!System.Net.Http.HttpClient..ctor()
  System.Net.Http.Functional.Tests.dll!System.Net.Http.Functional.Tests.DiagnosticsTest+<>c+<<SendAsync_HttpTracingEnabled_Succeeds>b__3_1>d.MoveNext()
  System.Private.CoreLib.dll!System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[[System.__Canon, System.Private.CoreLib]](System.__Canon ByRef)
  System.Net.Http.Functional.Tests.dll!System.Net.Http.Functional.Tests.DiagnosticsTest+<>c.<SendAsync_HttpTracingEnabled_Succeeds>b__3_1()
  System.Net.Http.Functional.Tests.dll!System.Diagnostics.Tracing.TestEventListener+<RunWithCallbackAsync>d__11.MoveNext()
  System.Private.CoreLib.dll!System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[[System.__Canon, System.Private.CoreLib]](System.__Canon ByRef)
  System.Net.Http.Functional.Tests.dll!System.Diagnostics.Tracing.TestEventListener.RunWithCallbackAsync(System.Action`1<System.Diagnostics.Tracing.EventWrittenEventArgs>, System.Func`1<System.Threading.Tasks.Task>)
  System.Net.Http.Functional.Tests.dll!System.Net.Http.Functional.Tests.DiagnosticsTest+<>c+<<SendAsync_HttpTracingEnabled_Succeeds>b__3_0>d.MoveNext()
  System.Private.CoreLib.dll!System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1[[System.Int32, System.Private.CoreLib]].Start[[System.__Canon, System.Private.CoreLib]](System.__Canon ByRef)
  System.Net.Http.Functional.Tests.dll!System.Net.Http.Functional.Tests.DiagnosticsTest+<>c.<SendAsync_HttpTracingEnabled_Succeeds>b__3_0()
```

**Comment by stephentoub** [comment]
I believe I see the problem.

**Comment by karelz** [comment]
Reopening to port the fix into rel/2.0.0 -- it is rather impactful on our test runs (4 failures in 1 day)

**Comment by karelz** [comment]
Fixed by #20236
